core/src/main/java/com/linecorp/armeria/client/retry/RetryConfig.java

```diff
@@ -0,0 +1,162 @@
+     * Returns config's retryRule, could be null.
+     */
+    @Nullable
+    public RetryRule retryRule() {
+        return retryRule;
+    }
+
+    /**
+     * Returns config's retryRuleWithContent, could be null.
+     */
+    @Nullable
+    public RetryRuleWithContent<T> retryRuleWithContent() {
+        return retryRuleWithContent;
+    }
+
+    /**
+     * Returns config's retry rule that is converted from retryRuleWithContent, could be null.
+     */
+    @Nullable
+    public RetryRule fromRetryRuleWithContent() {
+        return fromRetryRuleWithContent;
+    }
+
+    /**
+     * Returns config's maxContentLength, which is non-zero only if a {@link RetryRuleWithContent} is used.
+     */
+    public int maxContentLength() {
+        return maxContentLength;
+    }
+
+    /**
+     * Returns whether a {@link RetryRuleWithContent} is being used.
+     */
+    public boolean needsContentInRule() {
+        return needsContentInRule;
+    }
+
+    /**
+     * Returns whether the associated requires response trailers.
+     */
+    public boolean requiresResponseTrailers() {
+        return needsContentInRule() ?
+               retryRuleWithContent().requiresResponseTrailers() : retryRule().requiresResponseTrailers();
+    }
+
+    RetryConfig(RetryRule retryRule, int maxTotalAttempts, long responseTimeoutMillisForEachAttempt) {
+        checkArguments(maxTotalAttempts, responseTimeoutMillisForEachAttempt);
+        this.retryRule = requireNonNull(retryRule);
+        this.maxTotalAttempts = maxTotalAttempts;
+        this.responseTimeoutMillisForEachAttempt = responseTimeoutMillisForEachAttempt;
+        needsContentInRule = false;
+        maxContentLength = 0;
+        retryRuleWithContent = null;
+        fromRetryRuleWithContent = null;
+    }
+
+    RetryConfig(
+            RetryRuleWithContent<T> retryRuleWithContent,
+            int maxContentLength,
+            int maxTotalAttempts,
+            long responseTimeoutMillisForEachAttempt) {
+        checkArguments(maxTotalAttempts, responseTimeoutMillisForEachAttempt);
+        this.maxContentLength = maxContentLength;
+        this.retryRuleWithContent = requireNonNull(retryRuleWithContent);
+        fromRetryRuleWithContent = RetryRuleUtil.fromRetryRuleWithContent(retryRuleWithContent);
+        this.maxTotalAttempts = maxTotalAttempts;
+        this.responseTimeoutMillisForEachAttempt = responseTimeoutMillisForEachAttempt;
+        needsContentInRule = true;
+        retryRule = null;
+    }
+
+    private static void checkArguments(int maxTotalAttempts, long responseTimeoutMillisForEachAttempt) {
+        checkArgument(
+                maxTotalAttempts > 0,
+                "maxTotalAttempts: %s (expected: > 0)",
+                maxTotalAttempts);
+        checkArgument(
+                responseTimeoutMillisForEachAttempt >= 0,
+                "responseTimeoutMillisForEachAttempt: %s (expected: >= 0)",
+                responseTimeoutMillisForEachAttempt);
+    }
+}
```

Comment:
Please moves these after the member fields, so that the initialization code comes before other member methods.

Reply:
Done